### ISSUE TITLE
Firefox hangs or gives warning about an unresponsive script on cryptowat.ch

### ISSUE THREAD

**Issue by zorates** [kkapsner/CanvasBlocker#84]
Since the last update 0.3.2 of CanvasBlocker, Firefox often hangs or gives a warning about an unresponsive script when opening a chart on cryptowat.ch, such as https://cryptowat.ch/coinbase/btcusd/12h

This is with fake readout API and cryptowat.ch on the ignore list.
As far as I can tell this wasn't an issue with versions of CanvasBlocker before 0.3.2.

Things that I have tried that did not fix the issue:

- Put cryptowat.ch in the whitelist. No difference.
- Put cryptowat.ch in the whitelist and select block mode 'allow only whitelist'. No difference.
- Limited the maximal fake size to as low as 100 as explained here #77 . No diference.

**Comment by kkapsner** [comment]
OK - I see what the problem is. They are drawing very much text in the canvas. Since the new introduced mode monitors these functions there is a slightly overhead on each function call - even when you are not using it - that has an enormous impact on that page.

I might have to rethink my way of monitoring the functions...

**Comment by zorates** [comment]
Thank you for addressing this issue in the 0.3.3 update.

While it's definitely improved now, it still has significantly higher CPU usage while naviagting the site (compared to pre 0.3.2 versions), and it still occasionally becomes unresponsive with a warning message. 

As I make heavy use of that charting website most days, I currently have to keep Canvasblocker disabled, which is really unfortunate.

**Comment by kkapsner** [comment]
Does whitelisting help now?

**Comment by zorates** [comment]
I had it whitelisted when I tried, and while it's definitely not as bad as before, there's sill very noticeable overhead. 

However, I tried again and put the domain under 'ignored list' instead, and somehow it feels better like this compared to when it's whitelisted. Am I seeing things or is there an explanation for this?

**Comment by kkapsner** [comment]
This behaviour does not make much sense to me... Can you please tel me all your settings for thisn addon?

**Comment by zorates** [comment]
Yeah I didn't expect it to make much sense. But when I put it in the ignore list, it behaves pretty much like it did before 0.3.2. When I put it in the whitelist however, the site is slower with CPU spikes and unresponsive scripts. 

My settings are nothing special:
![canvasblocker_settings](https://cloud.githubusercontent.com/assets/23384969/20305502/1e704e5e-ab36-11e6-8416-b7d2a5c38b30.png)

**Comment by kkapsner** [comment]
Settings seem fine to me.

Did notifications show up when you had it whitelisted? What did you put in the whitelist?

**Comment by zorates** [comment]
I have been using the cryptowat.ch site now with CanvasBlocker enabled and the domain in the ignore list, and it works perfectly fine again, just as it did pre 0.3.2.

When I put the domain cryptowat.ch in the whitelist however (by cutting and pasting it from the ignore list), the performance issues are back. I just tested it again, and Firefox hanged itself completely without me being able to even click on the stop script button, and I had to force kill the process. I was visiting this url when that happened https://cryptowat.ch/coinbase/btcusd/3d

I'm not making this up, the difference between ignore and whitelist is extremely noticeable. But for now I'm just happy it works again when ignoring the domain :)
